web/store/issue/module/issue.store.ts

```diff
@@ -368,66 +368,77 @@ export class ModuleIssues extends IssueHelperStore implements IModuleIssues {
     }
   };
 
-  addModulesToIssue = async (workspaceSlug: string, projectId: string, issueId: string, moduleIds: string[]) => {
+  /**
+   * change modules array in issue
+   * @param workspaceSlug 
+   * @param projectId 
+   * @param issueId 
+   * @param addModuleIds array of modules to be added
+   * @param removeModuleIds array of modules to be removed
+   */
+  changeModulesInIssue = async (
+    workspaceSlug: string,
+    projectId: string,
+    issueId: string,
+    addModuleIds: string[],
+    removeModuleIds: string[]
+  ) => {
     // keep a copy of the original module ids
-    const originalModuleIds = this.rootStore.issues.issuesMap[issueId]?.module_ids ?? [];
+    const originalModuleIds = this.rootStore.issues.issuesMap[issueId]?.module_ids
+      ? [...this.rootStore.issues.issuesMap[issueId].module_ids!]
+      : [];
     try {
       runInAction(() => {
-        // add the new issue ids to the module issues map
-        moduleIds.forEach((moduleId) => {
+        // remove the new issue id to the module issues map
+        removeModuleIds.forEach((moduleId) => {
+          update(this.issues, moduleId, (moduleIssueIds = []) => {
+            if (moduleIssueIds.includes(issueId)) return pull(moduleIssueIds, issueId);
+            else return moduleIssueIds;
+          });
+        });
+        // add the new issue id to the module issues map
+        addModuleIds.forEach((moduleId) => {
           update(this.issues, moduleId, (moduleIssueIds = []) => {
             if (moduleIssueIds.includes(issueId)) return moduleIssueIds;
             else return uniq(concat(moduleIssueIds, [issueId]));
           });
         });
-        // update the root issue map with the new module ids
-        update(this.rootStore.issues.issuesMap, [issueId, "module_ids"], (issueModuleIds = []) =>
-          uniq(concat(issueModuleIds, moduleIds))
-        );
-      });
-
-      const issueToModule = await this.moduleService.addModulesToIssue(workspaceSlug, projectId, issueId, {
-        modules: moduleIds,
       });
+      if(originalModuleIds){
+        // update the root issue map with the new module ids
+        let currentModuleIds = concat([...originalModuleIds], addModuleIds);
+        currentModuleIds = pull(currentModuleIds, ...removeModuleIds);
+        this.rootStore.issues.updateIssue(issueId, { module_ids: uniq(currentModuleIds) });
+      }
+
+      //Perform API calls
```

Comment:
Can we do these in parallel?

Reply:
https://app.plane.so/plane/projects/02c3e1d5-d7e2-401d-a773-45ecba45d745/issues/d6698d59-188f-483c-9520-a7e7edf00563 
created an issue for this